proposed/2022/Package-Source-Mapping-VS-Options.md

```diff
@@ -0,0 +1,145 @@
+
+The add window has two buttons: _Add_ and _Close_.
+
+* Add: Add button adds package source mapping to project only using selected sources, and the pop-up will disappear. 
+
+* Close: Will close pop-up without adding any package source mappings.
+
+Below are the mockups for the package source mappings page. The top left is the package source mappings page, and the bottom left is the popup that appears when the user clicks the add button.
+
+![Options 1](../../meta/resources/PackageSourceMapping/VSOptions.png)
+
+#### Package Sources Page
+
+I will add a feature that allows a user to pin default sources for all package source mappings that are not specified on the package source mapping page. There will be a pin button, that when clicked, creates a default mapping for all package IDs to the pinned source. A pin icon will appear next to all pinned sources.
+
+Below are mockups of the package sources page with the added pin feature.
+
+![Options 2](../../meta/resources/PackageSourceMapping/VSOptions1.png)
+
+### Technical Explanation
+
+#### Package Source Mappings Page
+
+When the user opens the package source mapping page in the options dialog, it will automatically display all previous package source mapping configurations. 
+
+When the user adds a package source mapping to their project, the NuGet config will be updated when the user clicks the okay button. The specified package ID will be mapped to the selected source(s) in the config. 
+
+**Example 1** 
+
+```xml
+<PackageReference Include="Microsoft.A" Version="1.0.0"/>
+<PackageReference Include="Newtonsoft.Json" Version="11.0.0"/>
+<PackageReference Include="Serilog" Version="11.0.0"/>
+```
+
+```xml
+<packageSourceMapping>
+    <packagesource key="nuget.org">
+        <package pattern="Serilog" />
+        <package pattern="Microsoft.Extensions.Options" />
+        <package pattern="Newtonsoft.Json" />
+    </packageSource>
+</packageSourceMapping>
+```
+
+**Result:**
+
+In this example, the user added `Serilog`, `Microsoft.A`, and `Newtonsoft.Json`, separately, on the package source mappings page and selected `nuget.org` as the source from the `ComboBox` for all of them. All three package source mappings will be configured to be installed from `nuget.org`. 
+
+If the user removes one of these package source mappings, it will be deleted from the configuration as well. For example, if the user removes `Serilog`, the above code becomes
+
+```xml
+<packageSourceMapping>
+    <packagesource key="nuget.org">
+        <package pattern="Microsoft.Extensions.Options" />
+        <package pattern="Newtonsoft.Json" />
+    </packageSource>
+</packageSourceMapping>
+```
+
+If the user clicks the _Clear_ button on the package source mappings options page, then all package sources will be removed from the configuration, and package source mapping will be disabled. 

```

Comment:
See other clear comments, but is the idea that clear clears the configuration for that source or the general configuration?